rhalphalib/function.py

```diff
@@ -5,21 +5,44 @@
 from .util import install_roofit_helpers
 
 
-class BernsteinPoly(object):
-    def __init__(self, name, order, dim_names=None, init_params=None, limits=None, coefficient_transform=None):
+def matrix_bernstein(n):
+    v = np.arange(n + 1)
+    bmat = np.einsum("l,lv,lv->vl", binom.outer(n, v), binom.outer(v, v), np.power(-1., np.subtract.outer(v, v)))
+    bmat[np.greater.outer(v, v)] = 0  # v > l
+    return bmat
+
+
+def matrix_chebyshev(n):
+    M = np.zeros((n + 1, n + 1))
+    for nth in range(1, n + 2):
+        _coef_basis = np.zeros(nth)
+        _coef_basis[-1] = 1
+        c = np.polynomial.Chebyshev(_coef_basis, domain=[0, 1])
+        p = c.convert(kind=np.polynomial.Polynomial)
+        M[nth-1, :nth] = p.coef
+    return M
+
+
+def matrix_poly(n):
+    return np.identity(n + 1)
+
+
+class BasisPoly(object):
```

Comment:
You've probably already discussed this approach with @nsmith-, but I wonder if it would make sense to make a `BasisPoly` abstract class that `BernsteinPoly` and `ChebyshevPoly` and `Poly` inherit from... Maybe that could be done in such a way as to prevent the backward-incompatible change?

Reply:
Actually, that was my original implementation, but it seemed neater in use to switch the basis through a kwarg rather than making a switch between different classes. 

Could still fix the backcomp with `BernsteinPoly(BasisPoly)` and overwriting the init. No strong opinion

Reply:
That seems reasonable to me. Let's see what @nsmith- says.

Reply:
Subclassing and overriding the init seems fine to me